cpp/include/cudf/io_types.h

```diff
@@ -23,27 +23,47 @@
  */
 #pragma once
 
+typedef enum 
+{
+  FILE_PATH,
+  BUFFER,
+  URL
+} resource_type;
+
+typedef struct 
+{  
+  char			*data;			/**< pointer to a host memory buffer	*/
+  size_t		size;			/**< size of the buffer, in bytes		*/
+} buffer_desc;
+
+typedef struct
+{
+  resource_type type;
+
+  const char	*path;			/**< file location						*/
+  buffer_desc	buffer;			/**< host memory buffer					*/
+  char			*object;		// URL path. currently unsupported
```

Comment:
We should remove URL path for now.

IMO, it's better to handle it in the python layer, or using an outside module. I don't really see much advantage in having network-related code within read_csv() right now.